File: backend/app/models/alert.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Alert(Base):
    __tablename__ = "alerts"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    stream_id: Mapped[str] = mapped_column(String(100), index=True, nullable=False)
    level: Mapped[str] = mapped_column(String(30), default="warning", nullable=False)
    category: Mapped[str] = mapped_column(String(50), nullable=False)
    message: Mapped[str] = mapped_column(Text, nullable=False)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)

File: frontend/src/api/alert.ts
import { apiGet, apiPost } from "./client";

export interface Alert {
  id: number;
  stream_id: string;
  level: string;
  category: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

export function listAlerts() {
  return apiGet<Alert[]>("/alerts");
}

export function createTestAlert() {
  return apiPost<Alert>("/alerts/test");
}
